### PR TITLE
Ignore benign NumPy `__array_wrap__` deprecation warning in CI

### LINKED ISSUE #6848: CI emits NumPy DeprecationWarning: `__array_wrap__` must accept context and return_scalar arguments
### What

CI emits new `DeprecationWarning`s from NumPy in the GRPO, RLOO and Distillation trainer tests:

```
tests/test_distillation_trainer.py: 24 warnings
tests/test_grpo_trainer.py: 51 warnings
tests/test_rloo_trainer.py: 51 warnings
  /__w/trl/trl/.venv/lib/python3.12/site-packages/numpy/_core/fromnumeric.py:45: DeprecationWarning: __array_wrap__ must accept context and return_scalar arguments (positionally) in the future. (Deprecated NumPy 2.0)
    return conv.wrap(result, to_scalar=False)
```

CI run: https://github.com/huggingface/trl/actions/runs/32414561958/job/96572632322

Affected test files:

- `tests/test_grpo_trainer.py` (51 warnings)
- `tests/test_rloo_trainer.py` (51 warnings)
- `tests/test_distillation_trainer.py` (24 warnings)

The tests themselves pass; for now this is warning noise only.

### When it started

First seen in the run for #6619 (`47d20ec6`), and absent from the run for the preceding commit `88b99c2c`: https://github.com/huggingface/trl/actions/runs/32397964455 (4 warnings, none from NumPy).

### Next steps

Investigate the root cause, then decide whether anything is actionable on the TRL side and/or should be reported upstream.

### PR DESCRIPTION
**Pull request by albertvillanova**
This PR silences the benign NumPy `__array_wrap__` `DeprecationWarning` emitted by the InternVL VLM tests in CI.

Fix #6848.

### Motivation

Since #6619 added `trl-internal-testing/tiny-InternVLForConditionalGeneration` to the VLM parametrizations, the GRPO, RLOO and Distillation tests emit 126 warnings per run:

```
tests/test_distillation_trainer.py: 24 warnings
tests/test_grpo_trainer.py: 51 warnings
tests/test_rloo_trainer.py: 51 warnings
  /__w/trl/trl/.venv/lib/python3.12/site-packages/numpy/_core/fromnumeric.py:45: DeprecationWarning: __array_wrap__ must accept context and return_scalar arguments (positionally) in the future. (Deprecated NumPy 2.0)
    return conv.wrap(result, to_scalar=False)
```

`InternVLProcessor.__call__` passes the `num_patches` torch tensor to `np.cumsum`. NumPy tries `Tensor.cumsum(axis=…, dtype=…, out=…)`, which raises `TypeError`, so it falls back to its legacy path and wraps the result through `Tensor.__array_wrap__`, whose signature still predates NumPy 2.0. The tests pass; there is nothing actionable on the TRL side, and we never pass a tensor to NumPy in these code paths.

Reported upstream: https://github.com/huggingface/transformers/issues/48177 (still present in 5.15.1 and on `main`). The torch side of the chain is https://github.com/pytorch/pytorch/issues/180657, open since April with several unmerged attempts, so the transformers fix is the realistic one.

### Solution

Suppress the warning through `filterwarnings` in `pyproject.toml`, documented like the other upstream warning filters and linked to the tracking and upstream issues, so the entry can be removed once a fixed `transformers` release is available in CI.

### Changes

- Add a `filterwarnings` entry for the NumPy `__array_wrap__` `DeprecationWarning`, with a comment explaining the cause, what triggers it, and links to the tracking and upstream issues

---

> [!NOTE]
> **Low Risk**
> Test-config only: adds a pytest warning filter with no runtime, security, or training-path changes.
> 
> **Overview**
> Silences a benign NumPy 2.0 `DeprecationWarning` (`__array_wrap__` signature) in pytest so InternVL VLM runs of GRPO, RLOO, and Distillation tests no longer flood CI.
> 
> The warning comes from transformers passing a torch tensor to `np.cumsum` in the InternVL processor (upstream, not actionable in TRL). The new `filterwarnings` entry in `pyproject.toml` is documented with tracking and upstream issue links so it can be dropped once a fixed transformers release is in CI.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 526ad84e816e216d3133a580d83819c19527c500. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

**Comment by albertvillanova** [comment]
Merging as hotfix.
